better described how gas is used when interacting with a smart contra… (#11997)
ethereum/ethereum-org-website: public/content/developers/docs/transactions/index.md

Validator keeps the tip **+0.000210 ETH**

![Diagram showing how unused gas is refunded](./gas-tx.png)
_Diagram adapted from [Ethereum EVM illustrated](https://takenobu-hs.github.io/downloads/ethereum_evm_illustrated.pdf)_

Any gas not used in a transaction is refunded to the user account.

### Smart contract interactions {#smart-contract-interactions}

> Gas is required for any transaction that alters the state of a smart contract.

Reviewer: This wording is a little redundant, as a "transaction" does commit alterations to the chain: https://ethereum.org/en/glossary/#transaction

By changing "interaction" to "transaction" we could probably just drop the rest:

```suggestion
Gas is required for any transaction that involves a smart contract.
```

> Therefore, calling a [`view`](https://docs.soliditylang.org/en/latest/contracts.html#view-functions) or [`pure`](https://docs.soliditylang.org/en/latest/contracts.html#pure-functions) function from an EOA will not require any gas. The underlying RPC call for this scenario is [`eth_call`](/developers/docs/apis/json-rpc#eth_call)

Reviewer: ```suggestion
Smart contracts can also contain functions known as [`view`](https://docs.soliditylang.org/en/latest/contracts.html#view-functions) or [`pure`](https://docs.soliditylang.org/en/latest/contracts.html#pure-functions) functions, which do not alter the state of the contract. As such, calling these functions from an EOA will not require any gas. The underlying RPC call for this scenario is [`eth_call`](/developers/docs/apis/json-rpc#eth_call)
```

> Smart contracts can also contain functions known as [`view`](https://docs.soliditylang.org/en/latest/contracts.html#view-functions) or [`pure`](https://docs.soliditylang.org/en/latest/contracts.html#pure-functions) functions, which do not alter the state of the contract. As such, calling these functions from an EOA will not require any gas. The underlying RPC call for this scenario is [`eth_call`](/developers/docs/apis/json-rpc#eth_call)

Reviewer 2: I feel like we could simplify this whole thing and say that any attempt at state change requires gas, but reading Ethereum does not.

Author: Yes agree it could be simplified but as this is developer doc I thought it make sense to be a bit more explanatory. And as calling `view` and `pure` function internally cost gas I think that make sense to keep the part explaining the those two things
However I agree that the underlying RPC call is probably not the most interesting/relevant thing

> However calling a `view` or `pure` function internally (ie: from the contract itself or from another contract) will cost gas.

Reviewer: ```suggestion
Unlike when accessed using `eth_call`, these `view` or `pure` functions are also commonly called internally (ie. from the contract itself or from another contract) which does cost gas.
```

Validator keeps the tip **+0.000210 ETH**

In case of an interaction with a smart contract there is 2 cases possible:

> 1. A `view` or `pure` function (as described in [solidity documentation](https://docs.soliditylang.org/en/v0.8.23/contracts.html#view-functions)) is called from an EOA. In that specific case, no gas will be paid by the caller and the result will be returned. The underlying RPC call for this scenario is [eth_call](https://ethereum.org/en/developers/docs/apis/json-rpc#eth_call)

Reviewer: I see where you're going with this but it doesn't seem appropriate for where it is on the page. I might break this into it's own h3 header, ie `### Smart contract interactions {#smart-contract-interactions}`

I would probably leave something similar to the comment you deleted as well though, just updated. Maybe "Gas is require for any transaction that alters the state of a smart contract" instead... weaving in the rest of the content. 

Also, the link to `eth_call` should be relative: `/developers/docs/apis/json-rpc#eth_call`, and consider wrapping the term itself in single-backticks to style it as code.

Author: Hey @wackerow, thanks for your feedback, I made changes accordingly.

I agree for the h3 header, but in that case I would suggest to rename the `### On gas {#on-gas}` to `### Regular transactions {#regular-transactions}`. Let me know what's your opinion is on this
